src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -484,6 +484,16 @@ internal enum Protocol { TCP, NP, None, Admin };
 
         internal bool IsSsrpRequired { get; private set; } = false;
 
+        /// <summary>
+        /// Constructs and initializes a data source <b>for testing purposes</b> that
+        /// is of the provided protocol.
+        /// </summary>
+        /// <param name="protocol">Protocol to set the instance to use.</param>
+        internal DataSource(Protocol protocol)
```

Comment:
Was this changed purely for testing? 
If yes, it might benefit to keep this change out of the clientX change and leverage the appropriate connection string. like tcp:server,port or np:localhost  to get the various protocols out. 
I understand that this is away from the principles of unit testing, but something to consider to avoid changing the rest of the driver, to allow an easier merge to `main`

Reply:
Yes, this change was made for testing.
I think for data source we're going to need to make a separate class for SqlClientX development. We will need a shared memory enum value for Protocol, but making a change like that in main would be very bad. The best way to do that would be to duplicate the data source class for SqlClientX usage. This also allows for moving parsing logic out of the data source class and into the data source parsing handler, allowing for data source class to be a mostly data-only class, allowing for better testing of it.